hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java

```diff
@@ -103,7 +109,12 @@ public boolean getBoolean(HazelcastProperty groupProperty) {
      * @throws NumberFormatException if the value cannot be parsed
      */
     public int getInteger(HazelcastProperty groupProperty) {
-        return Integer.parseInt(properties[groupProperty.getIndex()]);
+        int configuredValue = Integer.parseInt(properties[groupProperty.getIndex()]);
+        if (configuredValue < 0) {
```

Comment:
Seems to be a good idea, yes. We should not print a warning if a user explicitly defines a negative default value.